components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodTerminalService.kt

```diff
@@ -270,12 +276,23 @@ class GitpodTerminalService(project: Project) {
         @Suppress("ObjectLiteralToLambda")
         shellTerminalWidget.addListener(object : TerminalWidgetListener {
             override fun allSessionsClosed(widget: TerminalWidget) {
-                terminalServiceFutureStub.shutdown(
-                        TerminalOuterClass.ShutdownTerminalRequest.newBuilder()
-                                .setAlias(supervisorTerminal.alias)
-                                .build()
-                )
+                lifetime.launch {
+                    delay(5000)
+                    terminalServiceFutureStub.shutdown(
+                            TerminalOuterClass.ShutdownTerminalRequest.newBuilder()
+                                    .setAlias(supervisorTerminal.alias)
+                                    .build()
```

Comment:
**Suggestion(non-blocking)**

```suggestion
                    try {
                        terminalServiceFutureStub.shutdown(
                                TerminalOuterClass.ShutdownTerminalRequest.newBuilder()
                                        .setAlias(supervisorTerminal.alias)
                                        .build()
                        )
                    } catch (e: StatusRuntimeException) {
                        LOG.info("Could not shutdown terminal ${supervisorTerminal.alias} due to ${e.message}")
                    }
```

Reply:
Good idea! Let me check if I can catch it like that.

Reply:
✅  [Try-catch added](https://github.com/gitpod-io/gitpod/compare/bdd2bb67b20e37c1f8d7edb790512a2daaba5f47..bfdd9c3beeac644999acebff8d36a1d4d8a9c593).